### DISCUSSION TITLE
Better way to create images?

### DISCUSSION THREAD

**Discussion by leoalv** [joomla/joomla-cms#37341] [General]
I have created a script to create images of Joomla articles using GD libraries, the script generates images on the fly (that is, when the page is visited and the image derived from the main one does not exist).

I have also seen in some extensions that the creation of the image is done when saving the item in the administrator. in this case many times unnecessary images are created

So my questions are:
Is the operating logic of my script correct?
My script affects performance by querying if the image exists or creating it when it doesn't exist?

Additional question:
Is it correct to use `getimagesize()` to get the dimensions of images that are not added using the Joomla `media` field?

What concerns me is performance and security

**Comment by dgrammatiko** [comment]
@leoalv just use https://responsive-images.dgrammatiko.dev https://extensions.joomla.org/extension/responsive-images/ you don't have to reinvent the wheel. You will get responsive images with support to webp and avif and also proper, cache invalidation, server side caching and much more

**Reply by leoalv** [reply]
Thanks @dgrammatiko it is true I must not reinvent the wheel.

I am trying to make a gallery of images in masonry using custom fields, where only the main image should be selected and from there create the thumbnails for masonry.

I will try your plugin that I see magnificent, to create both the masonry with the articles in the category view and the gallery with custom fields.

But I still have my question: if it affects the performance to create the images while the page loads (the first time). Or it would be better to create it from the administrator using a plugin.

**Reply by dgrammatiko** [reply]
> But I still have my question: if it affects the performance to create the images while the page loads (the first time). Or it would be better to create it from the administrator using a plugin.

That plugin can create the set of images per image upload, so you don't have that performance bottleneck. It still will create sets of images for existing images in the server in the run time if there are not any already cached, but you can simulate that on your localhost env and upload the cached images...

**Reply by leoalv** [reply]
I'm testing the plugin and it seems to me the best I've seen so far for creating responsive images.
Here the results:
Create images on the fly. When visiting the category it creates the images that depend on `image_intro`, when visiting the article it creates the images that depend on `image_fulltext`.

for the intro I chose a 540x430 image and it generated 3 new images _200, _320 and _480, but in the DOM the SRC of the image is `media/cached-resp-images/images/office_1920.jpg?version=714...` this image does not exist

In upload (from the article editor) the plugin creates the images the same if the images are created on the fly (intro or fulltext) in both cases it creates the number of resolutions that are smaller than the actual size of the image.

When the article is saved it does not create the images, I have not tried what happens when deleting.

It's an excellent plugin, but for what I need I can't use it because I need to display images of various mixed resolutions 400x400, 400x600, 600x800 to create a masonry. It is possible to show for example 320, 480 and 768, but if these values ​​are changed either in their order or value in the lugin my code that requests for example office_480.jpg, it would be in 404 error. But your plugin has made me look beyond my nose to improve my reinvention of the wheel and create a portfolio in joomla article.

Thanks @dgrammatiko
